collector_client_http.go

```diff
@@ -71,37 +71,43 @@ func newHttpCollectorClient(
 	url.Path = collectorHttpPath
 
 	return &httpCollectorClient{
-		reporterID:    reporterID,
-		accessToken:   opts.AccessToken,
-		attributes:    attributes,
-		reportTimeout: opts.ReportTimeout,
-		url:           url,
-		converter:     newProtoConverter(opts),
+		reporterID:      reporterID,
+		accessToken:     opts.AccessToken,
+		attributes:      attributes,
+		reportTimeout:   opts.ReportTimeout,
+		reportingPeriod: opts.ReportingPeriod,
+		url:             url,
+		converter:       newProtoConverter(opts),
 	}, nil
 }
 
 func (client *httpCollectorClient) ConnectClient() (Connection, error) {
-	// The golang http2 client implementation doesn't support plaintext http2 (a.k.a h2c) out of the box.
-	// According to https://github.com/golang/go/issues/14141, they don't have plans to.
-	// For now, we are falling back to http1 for plaintext.
-	// In the future, we might want to add out own h2c implementation (see https://github.com/hkwi/h2c).
-	var transport http.RoundTripper
-	if client.url.Scheme == "https" {
-		transport = &http2.Transport{}
-	} else {
-		transport = &http.Transport{}
+	// Use a transport independent from http.DefaultTransport to provide sane
+	// defaults that make sense in the context of the lightstep client.
+	transport := &http.Transport{
```

Comment:
Where they differ significantly from `DefaultTransport`, how did you choose parameters? e.g., are you experiencing an issue fixed by `DisableCompression: true`?

Reply:
`DisableCompression` here only controls compression of the response body, but the client is mostly doing uploads and the collectors aren't returning anything in the response, so it's a no-op.
This is just a setting to indicate explicitly that there's no point using compression, I can add a comment if you want.

The thing is, the zero-value of `http.Transport` is different from `http.DefaultTransport`, so the previous code already had a different behavior (and actually wasn't working with HTTP proxies), here's the standard configuration:
```
var DefaultTransport RoundTripper = &Transport{
        Proxy: ProxyFromEnvironment,
        DialContext: (&net.Dialer{
                Timeout:   30 * time.Second,
                KeepAlive: 30 * time.Second,
                DualStack: true,
        }).DialContext,
        MaxIdleConns:          100,
        IdleConnTimeout:       90 * time.Second,
        TLSHandshakeTimeout:   10 * time.Second,
        ExpectContinueTimeout: 1 * time.Second,
}
```
https://golang.org/pkg/net/http/#RoundTripper

Reply:
Yes, I see that the behaviour is different. The context of my question had been the reference to `DefaultTransport` in the context, since it seemed like the proposed `Transport` was meant to be juxtaposed against `DefaultTransport` rather than `http.Transport{}`.

Do you mind adding that comment about `DisableCompression`?

I definitely agree re: adding timeouts, etc. 👍

Reply:
Done!